File: Extensions.Tests/ChopTests.fs
namespace Tests.Extensions

open System
open Stringier.Extensions
open Microsoft.VisualStudio.TestTools.UnitTesting

[<TestClass>]
type ChopTests() =

    [<TestMethod>]
    member _.Chop0 () =
        Assert.ThrowsException<ArgumentOutOfRangeException>(fun () -> chop 0 "hello world" |> ignore) |> ignore

    [<TestMethod>]
    member _.Chop1 () =
        CollectionAssert.AreEqual([|"h";"e";"l";"l";"o";" ";"w";"o";"r";"l";"d"|], chop 1 "hello world")

    [<TestMethod>]
    member _.Chop2 () =
        CollectionAssert.AreEqual([|"he";"ll";"o ";"wo";"rl";"d"|], chop 2 "hello world")

    [<TestMethod>]
    member _.Chop3 () =
        CollectionAssert.AreEqual([|"hel";"lo ";"wor";"ld"|], chop 3 "hello world")

    [<TestMethod>]
    member _.Chop4 () =
        CollectionAssert.AreEqual([|"hell";"o wo";"rld"|], chop 4 "hello world")

    [<TestMethod>]
    member _.Chop5 () =
        CollectionAssert.AreEqual([|"hello";" worl";"d"|], chop 5 "hello world")

    [<TestMethod>]
    member _.Chop6 () =
        CollectionAssert.AreEqual([|"hello ";"world"|], chop 6 "hello world")

    [<TestMethod>]
    member _.Chop7 () =
        CollectionAssert.AreEqual([|"hello w";"orld"|], chop 7 "hello world")

    [<TestMethod>]
    member _.Chop8 () =
        CollectionAssert.AreEqual([|"hello wo";"rld"|], chop 8 "hello world")

    [<TestMethod>]
    member _.Chop9 () =
        CollectionAssert.AreEqual([|"hello wor";"ld"|], chop 9 "hello world")

    [<TestMethod>]
    member _.Chop10 () =
        CollectionAssert.AreEqual([|"hello worl";"d"|], chop 10 "hello world")

    [<TestMethod>]
    member _.Chop11 () =
        CollectionAssert.AreEqual([|"hello world"|], chop 11 "hello world")

    [<TestMethod>]
    member _.Chop12 () =
        CollectionAssert.AreEqual([|"hello world"|], chop 12 "hello world")

File: Tests/Stringier/Functions/ChopTests.cs
using System;
using Xunit;

namespace Stringier.Functions {
	public class ChopTests {
		[Theory]
		[InlineData("hello world", 1, new[] { "h", "e", "l", "l", "o", " ", "w", "o", "r", "l", "d" })]
		[InlineData("hello world", 2, new[] { "he", "ll", "o ", "wo", "rl", "d" })]
		[InlineData("hello world", 3, new[] { "hel", "lo ", "wor", "ld" })]
		[InlineData("hello world", 4, new[] { "hell", "o wo", "rld" })]
		[InlineData("hello world", 5, new[] { "hello", " worl", "d" })]
		[InlineData("hello world", 6, new[] { "hello ", "world" })]
		[InlineData("hello world", 7, new[] { "hello w", "orld" })]
		[InlineData("hello world", 8, new[] { "hello wo", "rld" })]
		[InlineData("hello world", 9, new[] { "hello wor", "ld" })]
		[InlineData("hello world", 10, new[] { "hello worl", "d" })]
		[InlineData("hello world", 11, new[] { "hello world" })]
		[InlineData("hello world", 12, new[] { "hello world" })]
		public void Chop(String text, Int32 size, String[] expected) {
			Int32 i = 0;
			foreach (ReadOnlySpan<Char> chunk in Text.Chop(text, size)) {
				Assert.Equal(expected[i++], chunk.ToString());
			}
		}

		[Fact]
		public void Chop_Invalid() => Assert.ThrowsAny<Langly.ArgumentException>(() => Text.Chop("hello world", 0));
	}
}
